### ISSUE TITLE
Can't access the PSP/GAME folder on Xbox One (UWP)

### ISSUE THREAD

**Issue by Ploggy** [hrydgard/ppsspp#9840]
Due to the XBOX ONE sandbox I can't access the Game Folder through the File Browser.
Is it possible to re-direct the location of the PSP/GAME folders to a place I can access on Xbox One?

**Comment by hrydgard** [comment]
Unfortunately it has to be in the app's private data, otherwise we can't use regular file APIs on it but must use UWP's horrible async IO stuff. Well, theoretically could do the same that I did with ISO files, but it would be quite a bit of work. 

What would be a bit easier would be to make an option somewhere to copy all the contents of PSP/GAME to some other location, and back. Contributions accepted...  I don't even have an Xbox One to test on. And if I'm gonna get one, which I might, I'm waiting until I can get the Xbox One X, which arrives in November.

**Comment by Ploggy** [comment]
Thanks mate I hope you get one too :) for selfish reasons of course lol
BTW could you look into my other issue I posted earlier? dump log there if it helps?

Thanks mate.

**Comment by WodsonKun** [comment]
Did you tried Xbox Device Portal?

**Comment by WodsonKun** [comment]
@Ploggy @hrydgard Xbox Device Portal haves a File Explorer that can open AppData folder... From there, you can upload your games and homebrews to PPSSPP private data.

**Comment by Ploggy** [comment]
You can upload files but not folders.

**Comment by dominater01** [comment]
another thing is since theres no file under any folder thats blank you cant upload there they block uploading aka the cheats directory 

i had to convert a file explorer homebrew app to iso make a folder then upload cheats.db there (in the cheats folder not new folder) then it worked fine

**Comment by WodsonKun** [comment]
Can you create folders? I don't remember now...

**Comment by dominater01** [comment]
only with the file explorer homebrew for ppsspp 

you can make folders on the local hdd not the dev section

**Comment by Ploggy** [comment]
Yeah the File Explorer app is ok (it allowed me to install the Atari ST Emulaotr.. But its a nasty workaround.

**Comment by Ploggy** [comment]
I just had an idea.. Would it be possible to change the URL from the PPSSPP's own Homebrew Store (http://store.ppsspp.org/index.html) to a local Webserver? so I can choose apps from my PC and upload from there instead .? 

Wouldn't that bypass the problem?

**Comment by hrydgard** [comment]
Well, we already have a different mechanism for sharing games. You run one copy of PPSSPP on a PC, go to Settings/Tools/Remote Disc Streaming. Press Share. Then go to your Xbox and go to the same place but click Browse.

This will let you play the games that are on your PC's recent list. In the future we should make it possible to share entire folders I guess..

**Comment by Ploggy** [comment]
Hmm, Well. If you can add support to share folders as well as ISO's then that would be good. :)
In the mean time could you add Emulators to the Homebrew Store? Looking though the list there are a lots of Homebrew Games but no Homebrew Emulators?

**Comment by unknownbrackets** [comment]
In theory, sharing homebrew could be done, but it'd require changing the disc0:/ interface to all use FileLoaders for access, and adding APIs to list directory contents, etc.  I guess each would get cached separately, and it'd mean less "read-ahead."  I'd expect bad performance...

It might be easier to just send a zip over and extract it locally on the other side.  It probably wouldn't be hard to make streaming support that, or add an interface to enter a zip file URL for homebrew I guess (might be painful with a controller.)

Sharing a folder of ISOs is probably not that hard, although it means exposing the subfolders and browsing, I guess.

-[Unknown]

**Comment by dominater01** [comment]
there also needs to be a way to backup gamesaves because if you uninstall ppsspp on xbox one they delete and you cant download from the dev portal

**Comment by unknownbrackets** [comment]
That sounds like #1019.

-[Unknown]

**Comment by dominater01** [comment]
yes that would be awesome for ppsspp to have so we can atleast upload our saves there doesnt need to be synchronizing just an upload i'd be happy

**Comment by strikersix23** [comment]
Maybe Hrydgard can possible add a homebrew file explorer to the store. for ease of access

**Comment by strikersix23** [comment]
If there is a server i can upload a file explorer for him
